src/writer.js

```diff
@@ -185,6 +179,29 @@ define([
         });
     };
 
+    var createSetValues = function (dataTree, form, xmlWriter) {
+        function writeSetValue(setValue) {
+            xmlWriter.writeStartElement('setvalue');
+            xmlWriter.writeAttributeString('event', setValue.event);
+            xmlWriter.writeAttributeString('ref', setValue.ref);
+            xmlWriter.writeAttributeString('value', setValue.value);
+            xmlWriter.writeEndElement();
+        }
+
+        _.each(form.getSetValues(), function (setValue) {
+            writeSetValue(setValue);
+        });
+
+        dataTree.walk(function (mug, nodeID, processChildren) {
+            if(mug && mug.options.getSetValues) {
+                _.each(mug.options.getSetValues(mug), function(setValue) {
+                    writeSetValue(setValue);
+                });
+            }
+            processChildren();
+        });
```

Comment:
`vellum:ignore="retain"` will probably not work either since it now creates `Ignored` mugs for data, bind, and control elements. I suppose it should probably be extended to handle setvalue nodes to conform with this PR.

The previous implementation (before this PR) preserved the order of setvalue nodes, so I don't believe we've had many problems (caused by vellum) related to ordering in the past.